### PR TITLE
New Library Releases (tx.supports() integration, Common custom chains + enum Chain/HF, Util v7.1.0)

### PR DESCRIPTION
**Pull request by holgerd77**
Releases with finalized `london` support and other substantial updates.

### CODE REVIEW

**Comment by holgerd77** [comment]
Still WIP and some releases missing (block, blockchain and VM), already open for some early review though. 😄 

Would wait on this until Friday to be able to include the `london` mainnet block to `Common`, have added the `blocked` label.

**Comment by holgerd77** [comment]
Also //cc in @fvictorio from Hardhat here

**Comment by holgerd77** [comment]
Short update: I was initially planning to do a mixture of minor version and bugfix releases here depending on the significance of features added to some of the libraries, but I am just realizing - also due to the discussions along #1329 - that this addition of the `mainnet` (and others) `london` block numbers to `Common` (together with bumping the `Common` version on all of the other releases) is actually a pretty significant thing, since this marks the first releases with full `london` support (before  things like reading block data from mainnet via RPC and then deciding on if it's a pre or post london block would just not work yet).

So I will switch here to do all releases as minor releases - think this is very much justified - and will also more prominently point the `london` readiness out in the release notes of all libraries once I am updating the blocks.

I would target a Friday evening (CET time) release on this, so it would be great if the PR can already get some (partial) reviews so that we make sure we are ready by then. This would be important for Hardhat and others so that they can integrate early on.

**Comment by holgerd77** [comment]
One little request: please also take these kind of reviews seriously and do a thorough and conscientious review. While this is not doing substantial code changes code changes along such PRs are nevertheless pretty heavy, and a single misspelled updated version number can in the worst case ruin the whole range of releases. 🙂

**Comment by holgerd77** [comment]
Hmm, seems it will take another 1-2 days until there is a decision on a `london` `mainnet` block, so I guess we'll have to postpone the release until early next week (Monday or eventually Tuesday). Would nevertheless be good if someone can already start reviewing here so that we'll be ready by then.

**Review by ryanio — APPROVED** [review]
just reviewed, looks excellent! helpful additional documentation, comprehensive changelogs, and digging the consistency of the singular enum naming. couldn't even find any typos 😋

File: packages/tx/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,79 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 3.3.0 - 2021-07-02
+
+### Improved L2 Tx Support
+
+This tx release bumps the `Common` library dependency version to `v2.4.0` and is therefore assured to works with the reworked `Common.custom()` method which can be used for an easier instantiation of common custom chain instances for sending txs to a custom (L2) network.
```
Comment (emersonmacro) [inline packages/tx/CHANGELOG.md:13]: Very minor suggestion: `is therefore assured to work with`

Comment (ryanio) [reply packages/tx/CHANGELOG.md:13]:
```suggestion
This tx release bumps the `Common` library dependency version to `v2.4.0` and is therefore assured to work with the reworked `Common.custom()` method which can be used for an easier instantiation of common custom chain instances for sending txs to a custom (L2) network.
```

**Comment by holgerd77** [comment]
Ok, I've now added the finalized `london` block numbers here in 9236f91 and added a respective "Finalized London HF Support" section to the release notes of all libraries, I also included the missing release notes from #1333 (since the PR didn't get merged due to being an out-of-schedule release).

This is now open for a final review (please be a bit loose on eventual additional typos and things on that level). 😄 

It would also be good if you don't merge other PRs until this release has been done.

I would then wait for https://github.com/ethereum/eth1.0-specs/pull/223 to be merged and then do the release (likely tomorrow morning I would assume).

**Comment by holgerd77** [comment]
Client tests are actually failing for the PR, which is a bit annoying (with `UncaughtException: Error: Unable to resolve module [multiformats/hashes/sha2] from [/home/runner/work/ethereumjs-monorepo/ethereumjs-monorepo/node_modules/libp2p-crypto/src/keys/rsa-class.js]`). Might have to do with the `package-lock.json` file I added on root along this latest commit round.

I likely won't have the time to fix and would have a tendency to ignore on this PR (and merge by admin-merge), since this is clearly not relevant for the PR. If someone wants to investigate and has an easy fix feel free to push here though (make sure to then give an additional approval). Otherwise I guess we can tackle separately as well.

**Comment by holgerd77** [comment]
Another approval would be nice here, would like to do the releases in 2 hours or so.

**Comment by holgerd77** [comment]
Ah, removed the `package-lock.json` and this indeed fixed the CI run for now 😄, this should nevertheless be tackled at some point.

I also added an additional reference to #1330 in the block release notes, @ryanio great that you picked this up - I actually planned to do so as well and integrate in the release but went over it. Really nice, would have been a pity if this would have missed this release round.

Nevertheless: it would still be good if someone could short-term approve here. 😄 

(I would otherwise admin-merge since Ryan already reviewed the most part here and I very much planned these releases in for the morning, but would feel a bit uneasy on this)

**Comment by holgerd77** [comment]
Ok, I will then go the uneasy path and just merge here and do the releases. 🙃

**Comment by holgerd77** [comment]
Just released the following releases on npm:

- `ethereumjs-util` [v7.1.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/ethereumjs-util%407.1.0)
- `@ethereumjs/common` [v2.4.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fcommon%402.4.0)
- `@ethereumjs/tx`[v3.3.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Ftx%403.3.0)
- `@ethereumjs/block` [v3.4.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fblock%403.4.0)
- `@ethereumjs/blockchain` [v5.4.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fblockchain%405.4.0)
- `@ethereumjs/vm` [v5.5.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fvm%405.5.0)

**Review by jochem-brouwer — COMMENTED** [review]
Checked the `.ts` files and especially cross-checked the block numbers / fork hashes for the hardforks. Looks good!

**Pull request merged**